sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -520,12 +522,14 @@ String name() {
     List<TopicPartition> partitions =
         Preconditions.checkArgumentNotNull(source.getSpec().getTopicPartitions());
 
-    // Each source has a single unique topic.
     for (TopicPartition topicPartition : partitions) {
-      this.kafkaTopic = topicPartition.topic();
-      break;
+      this.kafkaTopics.add(topicPartition.topic());
```

Comment:
Instead of initializing `kafkaTopics` to an empty set and then appending to it, would it make sense to declare it and move the assignment to here?
```
this.kafkaTopics = partitions.stream().map(TopicPartition::topic).collect(Collectors.toSet());
```

Reply:
Done